src/main/java/com/google/sps/servlets/ReasonServlet.java

```diff
@@ -1,27 +1,65 @@
 package com.google.sps.servlets;
 
+import static com.google.cloud.spanner.TransactionRunner.TransactionCallable;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import com.google.gson.Gson;
 import java.io.IOException;
 import java.time.Instant;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that sends reason for use to /reason. (will change to write to database)*/
+/** Servlet that writes Audit to database*/
 @WebServlet("/reason")
 public final class ReasonServlet extends HttpServlet {
+
+  DatabaseClient dbClient;
+  String queryString;
+  String selectedDatabase;
+  private Constants constants = new Constants();
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
+    response.setContentType(constants.TEXT_TYPE);
+    String reason = request.getParameter("reason");
+    String account = request.getParameter("account");
+    String query = request.getParameter("query");
+    queryString = request.getParameter("queryString");
+    selectedDatabase = "example-db";
+    this.dbClient = DatabaseConnector.getInstance().getDbClient(selectedDatabase);
+    insertUsingDml(this.dbClient,reason,account,query,response);
+  }
+
+  private void insertUsingDml(DatabaseClient dbClient, String reason, String account, String query, HttpServletResponse response) {
     Instant timestamp = Instant.now();
     String timeString = timestamp.toString();
-    // long timestamp = System.currentTimeMillis();
-    String text = getParameter(request, "reason", "");
-    text = text + " Time: " +  timeString;
-    String[] words = text.split("\\s*,\\s*");
-    response.setContentType("text/html;");
-    response.getWriter().println(Arrays.toString(words));
+    dbClient
+    .readWriteTransaction()
+    .run(
+        new TransactionCallable<Void>() {
+            @Override
+            public Void run(TransactionContext transaction) throws Exception {
+              String sql =
```

Comment:
Might be a good place to use String formatting. You could use String.format(" VALUES ('%s', '%s', '%s', '%s')", account, query, reason, timeString)
All the %s stand for a String, then you pass them into the String.format. It might make it feel a little cleaner.

Reply:
Thank you for the suggestion, this definitely looks a lot cleaner!